content/asm_4.md

```diff
@@ -1,194 +1,303 @@
+For example:
 
 ```assembly
-calculateStrLength:
-		;; check is it end of string
-		cmp byte [rsi], 0
-		;; if yes exit from function
-		je exitFromRoutine
-		;; load byte from rsi to al and inc rsi
-		lodsb
-		;; push symbol to stack
-		push rax
-		;; increase counter
-		inc rcx
-		;; loop again
-		jmp calculateStrLength
+;; Increment the value of the rcx
+inc rcx
+
+;; Add the value of the rcx to rdx
+add rdx, rcx
 ```
 
-As you can understand by it's name, it just calculates length of INPUT string and store result in rcx register. First of all we check that rsi register doesn't point to zero, if so this is the end of string and we can exit from function. Next is lodsb instruction. It's simple, it just put 1 byte to al register (low part of 16 bit ax) and changes rsi pointer. As we executed cld instruction, lodsb everytime will move rsi to one byte from left to right, so we will move by string symbols. After it we push rax value to stack, now it contains symbol from our string (lodsb puts byte from si to al, al is low 8 bit of rax). Why we did push symbol to stack? You must remember how stack works, it works by principle LIFO (last input, first output). It is very good for us. We will take first symbol from si, push it to stack, than second and so on. So there will be last symbol of string at the stack top. Than we just pop symbol by symbol from stack and write to OUTPUT buffer. After it we increment our counter (rcx) and loop again to the start of routine.
+## Logical instructions
+
+The logical instructions are used to execute [logical operations](https://en.wikipedia.org/wiki/Boolean_algebra#Operations):
 
-Ok, we pushed all symbols from string to stack, now we can jump to exitFromRoutine return to _start there. How to do it? We have ret instruction for this. But if code will be like this:
+- `and` - Logical *and*. The instruction takes two operands and performs the logical *and* operation on them. The result is stored in the first operand.
+- `or` - Logical *or*. The instruction takes two operands and performs the logical *or* operation on them. The result is stored in the first operand.
+- `xor` - Logical *xor*. The instruction takes two operands and performs the logical *xor* operation on them. The result is stored in the first operand.
+- `not` - Logical *not*. The instruction takes two operands and performs the logical *not* operation on them. The result is stored in the first operand.
```

Comment:
```suggestion
- `and` - The instruction takes two operands and performs the logical *and* operation on them. The result is stored in the first operand.
- `or` - The instruction takes two operands and performs the logical *or* operation on them. The result is stored in the first operand.
- `xor` - The instruction takes two operands and performs the logical *xor* operation on them. The result is stored in the first operand.
- `not` - The instruction takes two operands and performs the logical *not* operation on them. The result is stored in the first operand.
```

Reply:
I would remove first part of the explanation, as you anyway explain in the second sentence what logical operation is performed. Also, the name of the instruction itself is quite self-explanatory. 

Also, consider to move the last sentence repeated in all instructions into a separate sentence underneath:

`The result of these operations is always stored in the first operand.`